nvtabular/tools/dataset_inspector.py

```diff
@@ -0,0 +1,191 @@
+        """
+        data[col] = {}
+        # Get dtype
+        data[col]["dtype"] = str(ddf_dtypes[col].dtype)
+
+        # If string, chane string for its len
+        if data[col]["dtype"] == "object":
+            # data[col]["dtype"] = "string"
+            data[col]["cardinality"] = ddf[col].nunique().compute()
+            ddf[col] = ddf[col].map_partitions(lambda x: x.str.len())
+            ddf[col].compute()
+
+        # Get cardinality for cat and label
+        elif col_type == "cat" or col_type == "label":
+            data[col]["cardinality"] = ddf[col].nunique().compute()
+
+        # If multihot, compute list content stats, and change list for its len
+        if data[col]["dtype"] == "list":
+            ddf._meta[col] = cudf.Series([np.array([0], dtype=ddf_dtypes[col].dtype.leaf_type)])[:0]
+            # If list content is string
+            if ddf_dtypes[col].dtype.leaf_type == "object":
+                data[col]["multi_min"] = (
+                    ddf[col].compute().list.leaves.applymap(lambda x: x.str.len()).min()
+                )
+                data[col]["multi_max"] = (
+                    ddf[col].compute().list.leaves.applymap(lambda x: x.str.len()).max()
+                )
+                data[col]["multi_avg"] = (
+                    ddf[col].compute().list.leaves.applymap(lambda x: x.str.len()).mean()
+                )
+            # If list content is a number
+            else:
+                data[col]["multi_min"] = ddf[col].compute().list.leaves.min()
+                data[col]["multi_max"] = ddf[col].compute().list.leaves.max()
+                data[col]["multi_avg"] = ddf[col].compute().list.leaves.mean()
+            # Get list len for min/max/mean entry computation
+            # TODO: Add when cudf support is implemented
+            # https://github.com/rapidsai/cudf/issues/7157
+            # ddf[col] = ddf[col].map_partitions(lambda x: x.list.len(),
+            #                                    meta=(col, ddf_dtypes[col].dtype))
+            # ddf[col].compute()
+
+        # Get max/min/mean for all but label
+        if col_type != "label":
+            data[col][key_names["min"][col_type]] = ddf[col].min().compute()
+            data[col][key_names["max"][col_type]] = ddf[col].max().compute()
+            if col_type == "cont":
+                data[col][key_names["mean"][col_type]] = ddf[col].mean().compute()
+            else:
+                data[col][key_names["mean"][col_type]] = int(ddf[col].mean().compute())
+
+        # For conts get also std
+        if col_type == "cont":
+            data[col]["std"] = ddf[col].std().compute()
+
+        # Get percentage of nan for all
+        data[col]["per_nan"] = 100 * (1 - ddf[col].count().compute() / len(ddf[col]))
+
+    def inspect(self, path, dataset_format, columns_dict, output_file):
+        """
+        Parameters
+        -----------
+        path: str, list of str, or <dask.dataframe|cudf|pd>.DataFrame

```

Comment:
Why not use a nvtabular.Dataset object here? That wraps all the functionality you need (takes dask dataframe/path/cudf dataframe/ pandas dataframe)

Reply:
Dataset is the nvtabular.io.Dataset i think based on imports.

Reply:
The paths get turned into a nvt dataset. We ask for the paths so that the class can be used from command line to generate a json file.

Reply:
We are using the Dataset object here: `dataset = Dataset(path, engine=dataset_format)`.

Reply:
Whats the advantage here to taking a path/format and then immediately converting to a Dataset? Can' we just pass the dataset object in directly?